scapy/sendrecv.py

```diff
@@ -64,143 +96,98 @@ def sndrcv(pks, pkt, timeout = None, inter = 0, verbose=None, chainCC=0, retry=0
     hsent={}
     for i in tobesent:
         h = i.hashret()
-        if h in hsent:
-            hsent[h].append(i)
-        else:
-            hsent[h] = [i]
+        hsent.setdefault(i.hashret(), []).append(i)
+
     if retry < 0:
         retry = -retry
-        autostop=retry
+        autostop = retry
     else:
-        autostop=0
-
+        autostop = 0
+
+    if WINDOWS:
```

Comment:
Could we move this piece of code in a dedicated function, for example in scapy/arch ? This will make sndrcv() smaller and easier to read.

Reply:
`_get_pkt()` has specific implementations:
  * under windows (this one)
  * (but also) when using BPF
  * (or) when using PCAP under Darwin, FreeBSD or OpenBSD, and the socket is not a `SuperSocket` instance.

I don't think it would make the code easier to read to move only this implementation elsewhere. And we cannot move the last specific implementation anyway, since it relies on the socket type.